Playbooks: Terminology generalization (#4716)
mattermost/docs: source/incident-collaboration/launching-playbooks.rst

- ``/playbook check [checklist item]`` - Check/uncheck the specified task.
- ``/playbook announce ~[channels]`` - Announce the current playbook run in other channels.
- ``/playbook list`` - List all your ongoing playbook runs.
- ``/playbook owner [@username]`` - Show or change the current playbook run owner.
- ``/playbook info`` - Show a summary of the current playbook's activities.

To run a playbook use the ``/playbook start`` slash command from any channel. You can also use the desktop or browser to:

 * Select the **Playbook** icon in the channel header, and select **Run playbook** from the right-hand sidebar.
 * Use the context menu of a post and select **Run playbook**.

Running a playbook opens an interactive dialog. You can create a new playbook or select an existing playbook. Provide a name for your playbook run, then select **Start run**. Only playbooks you belong to are listed in the **Playbook** drop-down menu. The creator of a playbook run is automatically added as the first member and becomes the owner. Lastly, when the playbook run is created, it's announced in the associated channel by the Playbook Bot.

Joining a playbook run
~~~~~~~~~~~~~~~~~~~~~~

> Select the **Playbook** icon in the top menu bar to see a list of runs in progress. If the channel is public, you may join the channel without permission. You can also search for and join the channel via **Browse Channels**.

Author: ```suggestion
Select the **Playbook** icon in the top menu bar to see a list of runs, to which you have access, in progress. If the channel is public, you may join the channel without permission. You can also search for and join the channel via **Browse Channels**.
```

Running Playbooks
=================

Use slash commands to initiate playbook runs, make announcements, and get information about a playbook run.

Slash commands
~~~~~~~~~~~~~~

The ``/playbook`` slash command allows interaction with incidents via the post textbox on desktop, browser, and mobile for:

> - ``/playbook run`` - Start a playbook run.

Reviewer: We ended up keeping this as `/playbook start`. Perhaps something to iterate on, though?

> - ``/playbook run`` - Start a playbook run.

Author: ```suggestion
- ``/playbook start`` - Start a playbook run.
```

- ``/playbook end`` - End a playbook run.
- ``/playbook update`` - Update the current playbook run's status.
- ``/playbook restart`` - Restart an ended playbook run.
- ``/playbook check [checklist item]`` - Check/uncheck the specified task.
- ``/playbook announce ~[channels]`` - Announce the current playbook run in other channels.

> - ``/playbook list`` - List all your running playbooks.

Reviewer 2: Is this 100% correct? Because one can have multiple instances of the same playbook running. What about using `ongoing` or something similar?

```suggestion
- ``/playbook list`` - List all your ongoing playbook runs.
```

- ``/playbook owner [@username]`` - Show or change the current playbook run owner.
- ``/playbook info`` - Show a summary of the current playbook's activities.

To run a playbook use the ``/playbook start`` slash command from any channel. You can also use the desktop or browser to:

 * Select the **Playbook** icon in the channel header, and select **Run playbook** from the right-hand sidebar.
 * Use the context menu of a post and select **Run playbook**.

Running a playbook opens an interactive dialog. You can create a new playbook or select an existing playbook. Provide a name for your playbook run, then select **Start run**. Only playbooks you belong to are listed in the **Playbook** drop-down menu. The creator of a playbook run is automatically added as the first member and becomes the run owner.

> When the playbook run is created, it's announced in the configured channel. If that channel is archived or deleted, the Playbook bot posts a notification in the incident channel.

Reviewer: Was already here, but this paragraph feels out of place, only because the announcing has to be explicitly configured as an action, and I actually don't think there's any "fallback" behaviour for what happens if that channel is archived or deleted.

Reviewer 2: I think the paragraph is indeed correct (assuming the automation has been configured), as any error in the broadcasting mechanism would make the bot post an error message in the channel, but I also agree that it feels out of place. We may simply remove it?

Author: Happy to remove it, especially if its inclusion causes confusion.

Joining a playbook run
~~~~~~~~~~~~~~~~~~~~~~

> Select the **Playbook** icon in the top menu bar to see a list of runs in progress. If the channel is public, you may join the channel without permission. You can also search for and join the channel via **Browse Channels**.

Reviewer: Unfortunately, the behaviour here is to only list the runs to which you already have access. To join a run in progress, we should send the user to the backstage runs list.

> If the channel is private, an existing member must invite you. System Admins may join private playbook runs without permission.

Reviewer: This is true, but even system admins won't "see" private playbook runs, so it amounts to just saying that system admins can join private channels -- maybe omit?

> If the channel is private, an existing member must invite you. System Admins may join private playbook runs without permission.

Author: ```suggestion
If the channel is private, an existing member must invite you.
```

.. image:: ../images/IC-ad-hoc-tasks.gif

Changing owners
~~~~~~~~~~~~~~~

> To change playbook owners you can run the ``/playbook owner @username`` slash command from the playbook's channel. You can also use the desktop or browser to:

Reviewer 2: I think this should be `run`, right?

```suggestion
To change playbook owners you can run the ``/playbook owner @username`` slash command from the run's channel. You can also use the desktop or browser to:
```

1. Select the **Playbooks** icon in the channel header to open the right-hand sidebar.
2. Select the **Summary** tab.
3. Within the **Summary** tab, select the current owner's name in the right-hand sidebar, then select the new owner.
  * To change the owner to a user who is not in the channel, first add the user to the channel.

Providing a status update
~~~~~~~~~~~~~~~~~~~~~~~~~

Status updates ensure that stakeholders remain informed about the playbook run's progress. To post a status update:

1. Select the **Playbook** icon in the channel header to open the right-hand sidebar.

> 2. Find the playbook you're looking for and select **Go to Playbook**.

Reviewer 2: The button reads `Go to channel`
![image](https://user-images.githubusercontent.com/3924815/125490774-8c484287-a45e-46df-bce9-c304fba372b5.png)
```suggestion
2. Find the playbook you're looking for and select **Go to channel**.
```

> 3. Select the **Summary** tab.
> 4. Within the **Summary** tab, select **Update Status**.

Reviewer 2: Using the copy in the UI:

```suggestion
3. Select the **About** tab.
4. Within the **About** tab, select **Update status**.
```

By the way, after the Simplified RHS Design epic, the tabs will disappear, but that's only coming in August.

 * Status updates are posted to the incident channel as a message from the user providing the status update.
 * If the playbook has a defined broadcast channel, status updates are copied to the broadcast channel as a message from the incident bot.

> The most recent status post will also appear in the right-hand sidebar of the playbook channel. To correct or remove a status post, edit or delete the post as needed. Status updates that are broadcast to another channel will not be edited or deleted.

Reviewer 2: I *think* this should be also `run`, or `playbook run`.
```suggestion
The most recent status post will also appear in the right-hand sidebar of the run channel. To correct or remove a status post, edit or delete the post as needed. Status updates that are broadcast to another channel will not be edited or deleted.
```

> Ending playbook run

Reviewer: Should this be:
```suggestion
Ending a playbook run
```

~~~~~~~~~~~~~~~~~~~

Playbook run members can end an incident using the ``/playbook end`` slash command. Playbook runs can also be ended from the desktop or browser:

1. Select the **Playbook** icon in the channel header to open the RHS.

> 2. Find the incident you're looking for and select **Go to channel**.

Reviewer 2: ```suggestion
2. Find the playbook run you're looking for and select **Go to channel**.
```

3. Select **Update Status**.
4. From the **Status** drop-down menu, select **Resolved**.
5. Enter a message with additional details.
6. Select **Update Status**.

Ending a playbook run signals to all members of the channel that the playbook run is complete. Members of the team can continue to post in the channel, mark tasks as complete, and change the owner if needed.

> Restarting playbook run

Reviewer: Should this be:
```suggestion
Restarting a playbook run
```

~~~~~~~~~~~~~~~~~~~~~~~

> If playbook run was ended prematurely, it can be restarted within the channel using the ``/playbook restart`` slash command. Runs can also be restarted from the desktop or browser:

Reviewer 2: Not sure

```suggestion
If a playbook run was ended prematurely, it can be restarted within the channel using the ``/playbook restart`` slash command. Runs can also be restarted from the desktop or browser:
```
